### PR TITLE
Created add/edit pollPage

### PR DESCRIPTION
**Pull request by vikash-kanauja**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -43,9 +47,9 @@ function App() {
           element={<PrivateRoute Component={Users} redirectTo="/users" />}
         />
         <Route
-          path="/editPoll/:id"
+          path="/addedit"
           element={
-            <PrivateRoute Component={EditPollList} redirectTo="/editPoll" />
+            <PrivateRoute Component={AddEditPollPage} redirectTo="/addedit" />
```
Comment (yuvrajKaexpertise) [inline src/App.js]: delete these route

File: src/pages/AddEditPollPage.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  addPoll,
+  getSinglePoll,
+  updatePoll,
+  updatePollTitle,
+} from "../redux/reducers/pollListReducer";
+import { validateAddEditForm } from "../utils/validation";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
```
Comment (yuvrajKaexpertise) [inline src/pages/AddEditPollPage.js]: 3rd party import should be from top @vikash-kanauja

File: src/pages/AddEditPollPage.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+import Modal from "../Components/Modal";
+
+import {
+  addOption,
+  deleteOption,
+  updateOption,
+} from "../redux/reducers/optionReducer";
+import { MdDelete } from "react-icons/md";
+import { FaEdit } from "react-icons/fa";
```
Comment (yuvrajKaexpertise) [inline src/pages/AddEditPollPage.js]: these too @vikash-kanauja

File: src/pages/AddEditPollPage.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+const AddEditPoll = () => {
```
Comment (yuvrajKaexpertise) [inline src/pages/AddEditPollPage.js]: function name should always start with smallcase @vikash-kanauja

File: src/pages/AddEditPollPage.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+  const { id } = useParams();
+  const { state } = useLocation();
+  const [newPollData, setNewPollData] = useState({
+    title: "",
+    optionTitle: "",
+  });
+  const [options, setOptions] = useState([]);
+  const [errors, setErrors] = useState({ title: "", optionTitle: "" });
+  const [showModal, setShowModal] = useState(false);
```
Comment (yuvrajKaexpertise) [inline src/pages/AddEditPollPage.js]: which modal @vikash-kanauja ?

File: src/pages/AddEditPollPage.js [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+        }
+      }
+      setEditOption(null);
+      setNewPollData({ ...newPollData, optionTitle: "" });
+    } else {
+      setErrors(newErrors);
+    }
+  };
+
+  const handleDeleteOption = () => {
```
Comment (yuvrajKaexpertise) [inline src/pages/AddEditPollPage.js]: what is handle handle handle every where? addPoll , deletePoll function name not enough? @vikash-kanauja

File: src/pages/AddEditPollPage.js [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+    const newPoll = {
+      title: newPollData.title,
+      options,
+    };
+    const { newErrors, isVallid } = validateAddEditForm({
+      options,
+      title: newPollData.title,
+    });
+    if (isVallid) {
+      let result = {};
```
Comment (yuvrajKaexpertise) [inline src/pages/AddEditPollPage.js:150]: why do you need this result variable, if you are not using it @vikash-kanauja ?

**Review by yuvrajKaexpertise — CHANGES_REQUESTED** [review]

File: src/pages/AddEditPollPage.js [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+} from "../redux/reducers/optionReducer";
+
+const AddEditPollPage = () => {
+  const { id } = useParams();
+  const { state } = useLocation();
+  const [newPollData, setNewPollData] = useState({
+    title: "",
+    optionTitle: "",
+  });
+  const [options, setOptions] = useState([]);
```
Comment (amrit56789) [inline src/pages/AddEditPollPage.js]: which options?

File: src/pages/AddEditPollPage.js [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+  const [errors, setErrors] = useState({ title: "", optionTitle: "" });
+  const [showAddEditModal, setShowModal] = useState(false);
```
Comment (amrit56789) [inline src/pages/AddEditPollPage.js]: showAddEditModal and setShowModal ?????

File: src/pages/AddEditPollPage.js [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+  const [showDeletedModal, setShowDeletedModal] = useState(false);
+  const [selectedIndex, setSelectedIndex] = useState(null);
```
Comment (amrit56789) [inline src/pages/AddEditPollPage.js]: which selectedIndex??

File: src/pages/AddEditPollPage.js [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+  const [editOption, setEditOption] = useState(null);
```
Comment (amrit56789) [inline src/pages/AddEditPollPage.js]: which editOptions

File: src/pages/AddEditPollPage.js [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+  const { loading } = useSelector((state) => state.pollList);
```
Comment (amrit56789) [inline src/pages/AddEditPollPage.js]: which loading?

**Pull request merged**